components/armDecoder/encodings/armDataProcReg.cpp

```diff
@@ -378,11 +378,11 @@ arminst DP_3_SRC(armcode const &aFetchedOpcode, uint32_t aCPU, int64_t aSequence
   readRegister(inst, 2, rm, rs_deps[1], is_long ? false : sf);
 
   if (!is_high) {
-    predicated_action act =
-        addExecute(inst, operation(is_sub ? kSUB_ : kADD_), {kOperand3, kResult}, rs2_deps);
+    predicated_action act = addExecute(inst, operation(is_sub ? kSUB_ : kADD_),
```

Comment:
I was not able to determine really what this code is doing from reading it and from the commit message. Can we add a comment here for future knowledge/visibility?

Reply:
Ok, I will add more explanation in the commit message. This was indeed a bit complicated to find.

Originally, in MADD, the 'mul' action before this comment wrote its result to kResult, and then this 'act' action overwrote the kResult operand.
In cases where the 'act' action has to be re-executed because of mis-speculation, the original output of the 'mul' action is lost.
To solve this, I changed the output operand of 'act' to kResult1.